great_expectations/self_check/util.py

```diff
@@ -1278,47 +1280,39 @@ def get_test_validator_with_data(  # noqa: C901 - 31
-                # Further, some changes in pandas internal about how datetimes are created means to support pandas
-                # pre- 0.25, we need to explicitly specify when we want timezone.
-
-                # We will use timestamp for timezone-aware (UTC only) dates in our tests
-                if value.lower() in ["timestamp", "datetime64[ns, tz]"]:
-                    df[key] = df[key].str.strptime(pl_Datetime)
-                    continue
-                elif value.lower() in ["datetime", "datetime64", "datetime64[ns]"]:
-                    df[key] = df[key].str.strptime(pl_Datetime)
-                    continue
-                elif value.lower() in ["date"]:
-                    df[key] = df[key].str.strptime(pl_Date)
-                    value = "object"
-                try:
-                    type_ = np.dtype(value)
-                except TypeError:
-                    # noinspection PyUnresolvedReferences
-                    type_ = getattr(pl_datatypes, value)
-                    # If this raises AttributeError it's okay: it means someone built a bad test
-                polars_schema[key] = type_
-            # pandas_schema = {key: np.dtype(value) for (key, value) in schemas["pandas"].items()}
-            typed_df = pl_DataFrame()
-            for x in df.columns:
-                typed_df = typed_df.with_column(
-                    pl_lit(pl_Series(df[x].name, df[x].to_list()))
-                )
-                typed_df[x] = typed_df[x].cast(polars_schema[x], strict=False)
-
-            return build_polars_validator_with_data(df=typed_df)
-
-        if table_name is None:
-            # noinspection PyUnusedLocal
-            table_name = generate_test_table_name()
+            for (col_name, col_type) in schema.items():
+                # These names correspond to polars data types
+                if col_type == "Date":
+                    pl_df.replace(
+                        col_name,
+                        pl_df.get_column(col_name).str.strptime(
+                            pl.Date, fmt="%Y-%m-%d"
+                        ),
+                    )
+                elif col_type == "Datetime":
+                    pl_df.replace(
+                        col_name,
+                        pl_df.get_column(col_name).str.strptime(
+                            pl.Datetime, fmt="%Y-%m-%d %H:%M:%S"
+                        ),
+                    )
+                else:
+                    try:
+                        dtype = getattr(pl_datatypes, col_type)
+                    except AttributeError as e:
+                        raise AttributeError(
+                            f"{col_type} for column {col_name} is not a valid polars data type."
+                        ) from e
+
+                    pl_df.replace(
+                        col_name,
+                        pl_df.get_column(col_name).cast(dtype=dtype, strict=False),
+                    )
```

Comment:
Replacing columns with the correct data type. Polars doesn't infer a datetime from a string so we need to standardize on format in tests or we can alternatively use `dateutil.parser.parse()` to datetime.

Reply:
I want to think about this a little bit more... I'm running into an issue with the `in_set` tests right now with this, and it's because we create the validator (which goes through this conversion) even though we suppress the test ( funny, I know 😓)

Reply:
On second thought, I'm with you on this. I'd rather be explicit as well. I'll tag the tests that had to be modified in the `in_set` PR.. could you let me know what you think?